lambdas/service/app.py

```diff
@@ -737,204 +737,252 @@ def repository_search(entity_type: str, item_id: str):
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+
+    filter_schema = schema.object(**{
+        facet: schema.optional({
+            'oneOf': [
+                schema.object(**{'is': schema.array({})}),
+                *[
+                    schema.object(**{op: array_pair_spec})
+                    for op in ['contains', 'within', 'intersects']
+                ]
+            ]
+        })
+        for facet in facets
+    })
+    filter_schema['example'] = {'cellCount': {'within': [[10000, 1000000000]]}}
+    # Renders as "OrderedMap {}" in Swagger UI 3.24.2
+    filter_schema['default'] = {}
+    return params.query(
+        'filters',
+        schema.optional(application_json(filter_schema)),
+        description=format_description('''
+            Criteria to filter entities from the search results.
+
+            Each filter consists of a facet, an operator, and an array of facet
+            values. The available relationships are "is", "within", "contains",
+            and "intersects". See the `sort` parameter for a list of filter-able
+            facets.
+
+            For the "is" operator, multiple facet values are combined using "or"
+            logic. For example, `{"fileFormat": {"is": ["fastq", "fastq.gz"]}}`
+            selects entities where the file format is either "fastq" or
+            "fastq.gz". For the other relationships, the facet vales must come
+            in nested pairs specifying upper and lower bounds, and multiple
+            pairs are combined using "and" logic. For example,
+            `{"donorCount": {"within": [[1,5], [5,10]]}}` selects entities whose
+            donor organism count falls within both ranges, i.e., is exactly 5.
+        '''),
+    )
+
+
+def repository_search_spec(entity_type):
+    id_spec_link = f'#operations-Repository-get_repository_{entity_type}s__{entity_type}_id_'
+    facets = sorted(valid_facets())
+    sort_default, order_default = sort_defaults[entity_type + 's']
+    return {
+        'summary': f'Search the {entity_type}s index for entities of interest.',
+        'tags': ['Repository'],
+        'parameters': [
+            filters_param_spec(facets),
+            params.query(
+                'size',
+                schema.optional(schema.with_default(int, default=10)),
```

Comment:
```suggestion
                schema.optional(schema.with_default(int, 10)),
```

Reply:
Also consider that `int, 10` is quite redundant. The type can often be derived from the default.